### ISSUE TITLE
Duplicate warning shows inconsistently with multiple files

### ISSUE THREAD

**Issue by awayken** [rossPatton/stylint#61]
I'm linting a site that's comprised of a lot of different Stylus files. One problem that I've started to notice is that the warning thrown by the `"duplicates": true` flag generates some false positives when I'm checking all my Stylus files at once. Unfortunately, this false positive behavior is inconsistent.

I use a specific pattern in nearly all my Stylus files:
```stylus
$block-name = block

.{$block-name} {
    // Rules go here.
}
```

Obviously, the `globalDupe` option would cause me to get a lot of warnings about `$block-name`, so I have that set to `false`.

```json
{
    "alphabetical": true,
    "borderNone": true,
    "brackets": true,
    "colons": false,
    "colors": true,
    "commaSpace": true,
    "commentSpace": true,
    "cssLiteral": false,
    "depthLimit": false,
    "duplicates": true,
    "efficient": true,
    "enforceBlockStyle": true,
    "enforceVarStyle": true,
    "extendPref": "extends",
    "globalDupe": false,
    "indentSpaces": 4,
    "leadingZero": true,
    "maxWarnings": 30,
    "maxWarningsKill": true,
    "mixed": true,
    "namingConvention": "lowercase-dash",
    "parenSpace": false,
    "placeholders": true,
    "quotePref": "double",
    "semicolons": false,
    "universal": true,
    "valid": true,
    "whitespace": true,
    "zeroUnits": true,
    "zIndexDuplicates": false,
    "zIndexNormalize": false
}
```

Here is the output from running Stylint three times consecutively:
```bash
D:\Sites\stylint [master +0 ~0 -0]> node index.js ../acx-aim/src/stylus -c ../acx-aim/stylint.json
Warning:  property is not valid
File: ../acx-aim/src/stylus/_base/_reset.styl
Line: 29: -ms-interpolation-mode: bicubic;


�  1 Warnings
D:\Sites\stylint [master +0 ~0 -0]> node index.js ../acx-aim/src/stylus -c ../acx-aim/stylint.json
Warning:  duplicate property or selector, consider merging
File: ../acx-aim/src/stylus/_accordian.styl
Line: 9: $block-name = accordian

Warning:  property is not valid
File: ../acx-aim/src/stylus/_base/_reset.styl
Line: 29: -ms-interpolation-mode: bicubic;

Warning:  duplicate property or selector, consider merging
File: ../acx-aim/src/stylus/_search.styl
Line: 8: $block-name = search


�  3 Warnings
D:\Sites\stylint [master +0 ~0 -0]> node index.js ../acx-aim/src/stylus -c ../acx-aim/stylint.json
Warning:  duplicate property or selector, consider merging
File: ../acx-aim/src/stylus/_alert.styl
Line: 8: $block-name = alert

Warning:  property is not valid
File: ../acx-aim/src/stylus/_base/_reset.styl
Line: 29: -ms-interpolation-mode: bicubic;


�  2 Warnings
```

You can see how Stylint sometimes tells me about duplicates and sometimes doesn't. I made no code changes between running the commands. The files it flags are never the same, and they never actually have the issue.

I tested using f2de050c2e1b8d6fbc157dc2cd221dd043a778f6.

**Comment by rossPatton** [comment]
thanks, i'll take a look tomorrow

i plan on trying to address as many of these issues as possible over the weekend

**Comment by rossPatton** [comment]
will try to get to this sunday, and then push up all the latest to npm

in the meantime, can you double check your other issues to make sure they've been resolved? i've fixed a number of bugs on the develop branc

**Comment by rossPatton** [comment]
actually, think i might have just resolved this. please check out develop and test

**Comment by awayken** [comment]
I pulled develop and tested, but I'm still seeing this issue. I also had a bunch of new, but incorrect, `property is not valid` errors. Do you want me to submit a ticket for the new invalid property issues I'm finding?

**Comment by rossPatton** [comment]
hmm, can you just post here? i would like to try and prevent regressions if i can

i was hoping to get something up to npm tonight, ideally, but it can wait

**Comment by awayken** [comment]
I'll try to do these in blocks to make it easier to debug. (Hopefully.)

First, did you tweak something with the `"brackets"` option? I had it set to `true` before, but I think that was a mistake on my part. However, it wasn't throwing warnings before, and it is now.

In any case, it may have not been working before and now it is. Or it was always working, and I'm insane.

**Comment by awayken** [comment]
::selection Regression:

```stylus
::selection {
    background: #000;
    color: #fff;
}
::-moz-selection {
    background: #000;
    color: #fff;
}
```

Output:
```bash
Warning:  property is not valid
File: ../temp-stylus/regression.styl
Line: 1: ::selection {

Warning:  property is not valid
File: ../temp-stylus/regression.styl
Line: 5: ::-moz-selection {
```

**Comment by awayken** [comment]
font-face regression:

```stylus
// Fails
@font-face{
}
// Passes
@font-face {
}
```

Output:
```bash
Warning:  property is not valid
File: ../temp-stylus/regression.styl
Line: 13: @font-face{
```

**Comment by awayken** [comment]
Attribute selector regression:

```stylus
// Fails
[hidden],
// Passes
input[hidden],
// Fails
[readonly],
// Passes
input[readonly] {
}
```

Output:
```bash
Warning:  property is not valid
File: ../temp-stylus/regression.styl
Line: 20: [hidden],

Warning:  property is not valid
File: ../temp-stylus/regression.styl
Line: 24: [readonly],
```

**Comment by awayken** [comment]
template element regression:

```stylus
// Fails
template {
    display: block;
}
```

Output:
```bash
Warning:  property is not valid
File: ../temp-stylus/regression.styl
Line: 30: template {
```

**Comment by rossPatton** [comment]
Well, i don't think template or the attribute selector issues are regressions. I don't think template was white-listed before. Clearly I still have a lot of work to do to make all the tests bullet-proof. All of this looks very fixable though.

I think i'll just wait a bit to push again to npm, wanna make sure I fix as much of this as possible

**Comment by awayken** [comment]
I had another issue that I'm not able to isolate on its own, so I'll wait to submit that until I know more about how it works.

**Comment by rossPatton** [comment]
these should all be fixed in 0.9.3

if not, i plan on making 0.9.4 a small build. its just the amount of issues were starting to pile up

**Comment by awayken** [comment]
:+1: All the regressions I was able to isolate are fixed. I'll have to keep working on the one I can't quite isolate and file another ticket for it.

**Comment by rossPatton** [comment]
awesome, thanks!
